compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -632,23 +632,58 @@ class FuseTransposeWithLinalgOpConsumer
   bool allowGeneralizing = false;
 };
 
-bool isUnaryElementwiseGeneric(linalg::GenericOp genericOp) {
-  if (genericOp.getNumDpsInputs() != 1 || genericOp.getNumDpsInputs() != 1 ||
-      !linalg::isElementwise(genericOp)) {
-    return false;
+static bool isOperandMapAffectedByTransposeMap(
+    AffineMap indexingMap, ArrayRef<int64_t> iterationSpacePermutation) {
+  int64_t prevIdx = -1;
+  for (auto result : indexingMap.getResults()) {
+    assert(isa<AffineDimExpr>(result) &&
+           "expected projected permutation indexing map");
+    int64_t idx =
+        iterationSpacePermutation[cast<AffineDimExpr>(result).getPosition()];
+    // Verify that the relative ordering of indices in the map remain the same.
+    // If not, then the transposition affects the access order for the given
+    // map (and associated operand).
+    if (idx <= prevIdx) {
+      return true;
+    }
+    prevIdx = idx;
   }
+  return false;
+}
 
-  // Skip transposes and broadcasts. Transposes make more sense to fuse
-  // rather than propagate through, and broadcasts are cheaper to transpose
-  // before broadcasting.
-  if (genericOp.getMatchingIndexingMap(genericOp.getDpsInputOperand(0)) !=
-      genericOp.getMatchingIndexingMap(genericOp.getDpsInitOperand(0))) {
-    return false;
+static OpOperand *
+getSingleTransposedInputOperand(linalg::GenericOp genericOp,
+                                ArrayRef<int64_t> iterationSpacePermutation) {
```

Comment:
nit: add a doc to the method. I understand that it is convenient to return `nullptr` when there are no such operand or more than one operand. But people won't know that without looking at the implementation details.